module-apps/application-settings/windows/network/SimCardsWindow.cpp

```diff
@@ -38,32 +38,90 @@ namespace gui
         case Store::GSM::SIM::NONE:
         case Store::GSM::SIM::SIM_FAIL:
         case Store::GSM::SIM::SIM_UNKNOWN:
+        case Store::GSM::SIM::SIM_NEED_PIN:
+        case Store::GSM::SIM::SIM_NEED_PUK:
+        case Store::GSM::SIM::SIM_LOCKED:
             simStr      = utils::translate("app_settings_network_sim_none");
```

Comment:
Will this render `none` for a card that is connected, but not unlocked yet? I think that in this case it still should be `SIM1`/`SIM2` and `NONE` reserved for the cases as in the line 59 and below.

Reply:
No, because we display the text based on "selectedSIM", which actually has 3 states: SIM1, SIM2, NONE

Reply:
If this assignment won't disrupt any other logic, then it's OK for me.

Reply:
Fixed to separate enum for selected SIM with 3 fields: SIM1, SIM2, NONE